### ISSUE TITLE
False positive on numpy.mgrid

### ISSUE THREAD

**Issue by Brow71189** [python/mypy#11185]
**Bug Report**

The following code triggers a mypy type checking failure:

```python
import numpy

numpy.mgrid[0:10:1j]
```
Mypy reports "Slice index must be an integer or None" on this code snippet.

Admittedly this is a special case, because ``numpy.mgrid`` does not get "called" with parenthesis but square brackets. But a complex number as the third argument to mgrid is valid code and has a special meaning (it makes ``mgrid`` create an array with this many entries). See also: https://numpy.org/doc/stable/reference/generated/numpy.mgrid.html

EDIT:

``numpy.mgrid`` also allows floats for the first two "slices", which also triggers a mypy error:

```python
import numpy

numpy.mgrid[-1.5: 1.5]
``` 

**To Reproduce**

Type check the above code snippet with mypy.

**Expected Behavior**

The above code should pass type checking.

**Actual Behavior**

Mypy reports "Slice index must be an integer or None".

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.7
- Operating system and version: Windows 10
- numpy version: 1.20.3

**Comment by ethanhs** [comment]
Duplicate of https://github.com/python/mypy/issues/2410
